htdocs/societe/class/societe.class.php

```diff
@@ -1553,6 +1553,18 @@ public function delete($id, User $fuser = null, $call_trigger = 1)
 				}
 			}
 
+			// Remove third party  TODO trigger for external action linked to these accounts
+			if (! $error)
+			{
+				$sql = "DELETE FROM ".MAIN_DB_PREFIX."societe_account";
```

Comment:
This is surely a bug. The rowid in this table is not same than rowid in societe.
May be you just wnt to add the table societe_account into the property array  ->childtablesoncascade

Reply:
@eldy fix done... really need a break :p